coral-hive/src/main/java/com/linkedin/coral/transformers/CoralRelNodeToCoralSqlNodeTransformer.java

```diff
@@ -0,0 +1,172 @@
+    List<String> tableNameWithoutCatalogAndDb = e.getTable().getQualifiedName().subList(1, 3);
+    final SqlIdentifier identifier = new SqlIdentifier(tableNameWithoutCatalogAndDb, SqlParserPos.ZERO);
+    return result(identifier, ImmutableList.of(Clause.FROM), e, null);
+  }
+
+  /**
+   * Correlate represents the FROM clause of the final sql query
+   * when the FROM clause contains explode() UDF as well
+   *
+   * Overriding the default implementation is required because
+   * the rightChildResult is also overriden and doesn't match super's expectation
+   */
+  @Override
+  public Result visit(Correlate e) {
+    //Left Result removes the catalog name "hive" from the table path
+    final Result leftResult = visitChild(0, e.getLeft());
+
+    // Add context specifying correlationId has same context as its left child
+    correlTableMap.put(e.getCorrelationId(), leftResult.qualifiedContext());
+
+    final Result rightResult = visitChild(1, e.getRight());
+
+    SqlJoin join = new SqlJoin(POS, leftResult.asFrom(), SqlLiteral.createBoolean(false, POS),
+        JoinType.COMMA.symbol(POS), rightResult.asFrom(), JoinConditionType.NONE.symbol(POS), null);
+
+    SqlIdentifier leftNode = (SqlIdentifier) leftResult.node;
+    SqlBasicCall lateralRightNode = (SqlBasicCall) rightResult.node;
+    SqlBasicCall asSqlNode = (SqlBasicCall) lateralRightNode.getOperands()[0];
+    SqlIdentifier asSqlNodeAlias = (SqlIdentifier) asSqlNode.getOperands()[1];
+
+    //The default implementation assumes that the rightLateralNode, for LateralSqlOperator,
+    // has a null alias. At the same time, it expects a non-null alias for the same node.
+    // At this point, we can either customize the rightLateralNode to pass the alias check
+    // or we can bypass the check and work around it. The latter appoach has been chosen.
+    final ImmutableMap.Builder<String, RelDataType> builder =
+        ImmutableMap.<String, RelDataType> builder().put(Util.last(leftNode.names), leftResult.neededType)
+            .put(Util.last(asSqlNodeAlias.names), rightResult.neededType);
+
+    return new Result(join, Expressions.list(Clause.FROM), null, null, builder.build());
+  }
+
+  /**
+   * CORAL represents explode function as an Uncollect RelNode.
+   * The default super's implementation takes Project RelNode as input in the form of
+   *        SELECT `complex`.`c` as`ccol` FROM (VALUES  (0))
+   * and appends the UNNEST and AS operators on top of it.
+   *
+   * However, the result generated is unmanageable for the following reasons:
+   * 1. Result has some extra clauses with make parsing difficult.
+   * 2. It also doesn't work well for unboxing array of type struct.
+   * It attempts to generate individual columns for each datatype inside the struct.
+   * 3. It doesn't append the LATERAL operator
+   *
+   * Overriding the function helps generate a more easily parsable SqlNode:
+   * LATERAL UNNEST(`complex`.`c`) AS t0 (ccol)
```

Comment:
I think the premise is just generating  original the Coral SqlNode, because that is the only way the conversion pipeline can work.

Reply:
that's a good point. 
I rephrased the javadoc and also added a line `Also, the above Result's SqlNode does not mimic the original SqlNode as expected.` Do you think this is sufficient context or should we describe the "original SqlNode" a bit here?